scrapper.py

```diff
@@ -25,39 +39,146 @@ class Crawler:
+            'a')['href'] for article_summary_bs in article_summaries_bs]
+        self.urls.extend(urls_bs)
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+
+        for url in self.seed_urls:
+            response = requests.get(url)
+            sleep_period = random.randint(1, 3)
+            sleep(sleep_period)
+
+            if not response.ok:
+                continue
+
+            soup = BeautifulSoup(response.text, 'lxml')
+            self._extract_url(soup)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        article_urls_bs = article_bs.find('a', class_='obj_galley_link pdf')['href']
+        response = requests.get(article_urls_bs)
+        soup = BeautifulSoup(response.text, 'lxml')
+        url_download_pdf = soup.find('a', class_='download')['href']
+        pdf = PDFRawFile(url_download_pdf, self.article_id)
+        pdf.download()
+
+        self.article.text = pdf.get_text().split('СПИСОК ЛИТЕРАТУРЫ')[0]
+
+    def _fill_article_with_meta_information(self, article_bs):
+        # title
+        self.article.title = article_bs.find('h1', class_='page_title').text.strip()
+
+        # author
+        self.article.author = article_bs.find('ul', class_='item authors').find('li').find('span').text.strip()
+
+        # topics
+        self.article.topics = article_bs.find('div', class_='item keywords').find(
+            'span', class_='value').text.strip().replace('\t', "").split(', ')
+
+        # date
+        try:
+            date = article_bs.find('div', class_='item published').find('div', class_='value').text.strip()
+            self.article.date = datetime.strptime(date, '%Y-%m-%d')
+        except AttributeError:
+            self.article.date = datetime.strptime('2021', '%Y')
+
+    def parse(self):
+        response = requests.get(url=self.article_url)
+
+        article_bs = BeautifulSoup(response.text, 'lxml')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+
+    as_path = Path(base_path)
+    if as_path.exists():
+        shutil.rmtree(as_path)
+    as_path.mkdir(parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as file:
+        config = json.load(file)
+
+    pattern = re.compile(r"https?://journal\.asu\.ru/urisl/issue/.+")
```

Comment:
it should be in a more configurable place. think about it and propose a better place

Reply:
I think I could keep it in constants.py